pkg/util/collatedstring/collatedstring.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package collatedstring
+
+import "golang.org/x/text/collate"
+
+// DefaultCollationTag is the "default" collation for strings.
+const DefaultCollationTag = "default"
+
+// CCollationTag is the "C" collation for strings. At the time of writing
+// this, it behaves the same os "default" since LC_COLLATE cannot be modified.
+const CCollationTag = "C"
+
+// PosixCollationTag is the "POSIX" collation for strings. At the time of
+// writing this, it behaves the same os "default" since LC_COLLATE cannot be
+// modified.
+const PosixCollationTag = "POSIX"
+
+var supportedTagNames []string
```

Comment:
should we use `language.Tag` here?

Reply:
hmm, maybe we can't because `language.Tag` has metadata in it. oh well!

Reply:
yeah i had tried to get it to work with `Tag`, but it can't. there's no way to even make a "custom" `Tag`.